### PR TITLE
Add E2E Tests

### LINKED ISSUE #1850: Support coverage for tests

### PR DESCRIPTION
**Pull request by danielr18**
Interesting to know:
- There's a function called `flushTriggers` that must be called in the tests after calling endpoints to ensure that all triggers have been processed before asserting conditions. The way this function works is by connecting to the logs websocket and waiting for no new logs for 2 seconds.
- Coverage support is added by aggregating coverage files that are created after each firebase function runs. These files need to be created because the coverage is collected for each individual function, so it can't rely on just one endpoint to return the global coverage.

Current coverage:
[coverage.zip](https://github.com/daostack/common-backend/files/11549285/coverage.zip)

Closes #1850

### CODE REVIEW

File: firebase.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
       "port": 8085
     },
     "hosting": {
-      "port": 5000
+      "port": 5001
```
Comment (ariel-dao) [inline firebase.json:29]: why change port?

File: functions/package.json [hunk 2]
```diff
@@ -47,34 +50,41 @@
...
     "uuid": "^9.0.0",
+    "ws": "^8.13.0",
     "yup": "^1.1.1"
   },
   "devDependencies": {
     "@types/chai": "^4.3.5",
     "@types/express": "^4.17.17",
     "@types/express-pino-logger": "^4.0.3",
     "@types/google-libphonenumber": "^7.4.23",
+    "@types/ioredis": "^5.0.0",
```
Comment (ariel-dao) [inline functions/package.json]: remove

File: firebase.json (hunk 1 above)
Comment (danielr18) [reply firebase.json:29]: MacOS uses Port 5000 for Control Center

**Comment by ariel-dao** [comment]
Getting error on `test-e2e`:
```
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Error during global setup ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Error: connect ECONNREFUSED 127.0.0.1:8086
 ❯ AxiosError.from node_modules/axios/lib/core/AxiosError.js:89:14
 ❯ RedirectableRequest.handleRequestError node_modules/axios/lib/adapters/http.js:591:25
 ❯ RedirectableRequest.emit node:events:513:28
 ❯ eventHandlers.<computed> node_modules/follow-redirects/index.js:14:24
 ❯ ClientRequest.emit node:events:513:28
 ❯ Socket.socketErrorListener node:_http_client:502:9
 ❯ Socket.emit node:events:513:28

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Serialized Error: Object {
  "address": "127.0.0.1",
  "code": "ECONNREFUSED",
  "config": Object {
    "adapter": [Array],
    "data": undefined,
    "env": [Object],
    "headers": [AxiosHeaders],
    "maxBodyLength": -1,
    "maxContentLength": -1,
    "method": "delete",
    "timeout": 0,
    "transformRequest": [Array],
    "transformResponse": [Array],
    "transitional": [Object],
    "url": "http://127.0.0.1:8086/emulator/v1/projects/common-dev-dea4e/accounts",
    "validateStatus": [Function validateStatus],
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
  },
  "errno": -111,
  "port": 8086,
  "request": Writable {
    "_currentRequest": [ClientRequest],
    "_currentUrl": "http://127.0.0.1:8086/emulator/v1/projects/common-dev-dea4e/accounts",
    "_ended": true,
    "_ending": true,
    "_events": [Object],
    "_eventsCount": 3,
    "_maxListeners": undefined,
    "_onNativeResponse": [Function anonymous],
    "_options": [Object],
    "_redirectCount": 0,
    "_redirects": [Array],
    "_requestBodyBuffers": [Array],
    "_requestBodyLength": 0,
    "_writableState": [WritableState],
    Symbol(kCapture): false,
  },
  "syscall": "connect",
}
Caused by: Error: connect ECONNREFUSED 127.0.0.1:8086
 ❯ TCPConnectWrap.afterConnect [as oncomplete] node:net:1494:16

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Serialized Error: {
  "address": "127.0.0.1",
  "code": "ECONNREFUSED",
  "errno": -111,
  "port": 8086,
  "syscall": "connect",
}
```

**Review by ariel-dao — CHANGES_REQUESTED** [review]
Overall looks good, but I'm getting errored out on e2e tests

**Comment by danielr18** [comment]
Did the `npm run dev`  command throw an issue?

**Review by ariel-dao — APPROVED** [review]

**Pull request merged**